receiver/azuremonitorreceiver/config.go

```diff
@@ -254,6 +254,7 @@ type Config struct {
 	AppendTagsAsAttributes            bool                          `mapstructure:"append_tags_as_attributes"`
 	UseBatchAPI                       bool                          `mapstructure:"use_batch_api"`
 	Dimensions                        DimensionsConfig              `mapstructure:"dimensions"`
+	MaximumResourcesPerBatch          int                           `mapstructure:"maximum_resources_per_batch"`
```

Comment:
Could we at least validate this isn't a negative number? I think it's reasonable to assume that's an invalid value for Azure, and we want to fail fast where possible.